calypso/contracts.go

```diff
@@ -124,6 +124,51 @@ func (c ContractWrite) Spawn(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instruc
 	return
 }
 
+// Invoke supports the following command:
+//  - update - it takes a 'data' and/or 'extraData' argument that is used to
+//    update the data and/or extradata part of the write structure.
+func (c *ContractWrite) Invoke(rst byzcoin.ReadOnlyStateTrie,
+	inst byzcoin.Instruction, cin []byzcoin.Coin) (sc []byzcoin.StateChange,
+	cout []byzcoin.Coin, err error) {
+	cin = cout
+
+	var darcID darc.ID
+	_, _, _, darcID, err = rst.GetValues(inst.InstanceID.Slice())
+	if err != nil {
+		return
+	}
+
+	update := false
+
+	switch inst.Invoke.Command {
+	case "update":
+		data := inst.Invoke.Args.Search("data")
+		if data != nil {
+			c.Data = data
+			update = true
+		}
+		extraData := inst.Invoke.Args.Search("extraData")
+		if extraData != nil {
+			c.ExtraData = extraData
+			update = true
+		}
+	default:
+		return nil, nil, xerrors.New("only know 'update' command")
+	}
+
+	if !update {
+		return
```

Comment:
Maybe an error? IMO, `update` without either `data` or `extraData` is usually a bug.

Reply:
Hmm - still thinking about it. Per se it's a NO-OP, so it's a valid operation. On the other hand, it might point to an error done somewhere in the sending of the transaction...